server/sandbox.go

```diff
@@ -360,11 +368,12 @@ func (s *Server) RemovePodSandbox(ctx context.Context, req *pb.RemovePodSandboxR
 		return nil, err
 	}
 
-	podInfraContainerName := sb.name + "-infra"
-	var podInfraContainer *oci.Container
+	podInfraContainer := sb.infraContainer
+	containers := sb.containers.List()
+	containers = append(containers, podInfraContainer)
```

Comment:
Your logic retains the existing flow so it is fine. I want us to move the stop/remove/delete of pod infra after the other containers. We can do that in a follow up PR.